src/NServiceBus.Core/Transports/Learning/LearningTransportDispatcher.cs

```diff
@@ -57,6 +59,14 @@ Task DispatchUnicast(IEnumerable<UnicastTransportOperation> operations, Transpor
 
         async Task WriteMessage(string destination, IOutgoingTransportOperation transportOperation, TransportTransaction transaction)
         {
+            var headerPayload = HeaderSerializer.Serialize(transportOperation.Message.Headers);
+            var headerSize = Encoding.UTF8.GetByteCount(headerPayload);
```

Comment:
The serializer is hardcoded to UTF8 anyway? https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Transports/Learning/HeaderSerializer.cs#L20

I'll add a unit test to cover this so that we'll be aware if we break?